cmd/registry_login_test.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func Test_GenerateRegistryAuth(t *testing.T) {
+	registryURL := "https://index.docker.io/v1/"
+	username := "docker_user"
+	password := "docker_password"
+	gotBytes, _ := generateRegistryAuth(registryURL, username, password)
+	got, err := bytesToRegistryStruct(gotBytes)
+
+	if err != nil {
+		t.Errorf("Error converting bytes to struct, %q", err)
+		t.Fail()
+	}
+
+	want := RegistryAuth{
+		AuthConfigs: map[string]Auth{
+			registryURL: {Base64AuthString: base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", username, password)))},
+		},
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Error("Structs were not equal, the generated bytes did not match the expected output")
+		t.Fail()
+	}
+}
+
+func Test_GenerateRegistryAuthNoRegistryURL(t *testing.T) {
+	registryURL := ""
+	username := "docker_user"
+	password := "docker_password"
+	_, err := generateRegistryAuth(registryURL, username, password)
+
+	if err == nil {
+		t.Error("We were expecting an error as 'registryURL' was empty, but we didnt get one")
```

Comment:
Good descriptive message, but what about `Want err, but got nil`? -> The title of the test already expresses the message. By keeping more concise it will be easier to maintain the tests over time.